isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+Die Konfiguration dieser Frameworks wird in ihren Nutzungsvorgaben beschrieben.
+
+[[beans-standardmaessig-als-singletons-definieren]]
+==== Beans standardmäßig als Singletons definieren
+
+Spring-Beans können entweder als Singletons mit nur einer Instanz, mit einer Instanz pro Aufruf oder mit einer Instanz pro Abhängigkeit (Prototype) erzeugt werden.
+Die Spring-Beans einer Anwendung sollen zustandslos sein und werden als Singleton-Beans erzeugt.
+Wo technisch erforderlich können auch andere Scopes verwendet werden.
+
+[[vorgaben-zur-spring-konfiguration]]
+==== Vorgaben zur Spring-Konfiguration
+
+Die Spring-Konfiguration der Anwendung ist nach folgenden Vorgaben zu erstellen.
+
+Der zentrale Ausgangspunkt für die Spring-Konfiguration ist die <<listing-applikationsklasse>>.
+Diese wird im Package der Anwendung (`<org>.<domäne>.<anwendung>`) erstellt.
+Sie ist mit `@Configuration` und `@EnableAutoConfiguration` annotiert.
+Damit wird die Autokonfiguration von Spring Boot eingeschaltet.
+Zusätzlich erbt die Klasse von `SpringBootServletInitializer`, damit die Anwendung im Tomcat deployt werden kann.
+Die Konfigurationsklassen der Schichten werden per `@Import`-Annotation eingebunden.
+
+[[listing-applikationsklasse]]
+.Applikationsklasse der Anwendung
+[source,java]
+----
+package de.beispiel.if2anwendung;
+
+@Configuration
+@EnableAutoConfiguration
+@Import({ CoreConfig.class, PersistenceConfig.class, ServiceConfig.class })
+public class IsyFactApplication extends SpringBootServletInitializer {
+
+    @Override
+    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+        return application.sources(IsyFactApplication.class);
+    }
+}
+----
+
+Für jede Schicht wird im Package der Schicht (`<org>.<domäne>.<anwendung>.<schicht>`) parallel zu den Interfaces der Komponentenschnittstellen eine mit `@Configuration` annotierte <<listing-konfigurationsklasse,Konfigurationsklasse>> erstellt.
+Diese wird zusätzlich mit `@ComponentScan` annotiert.
+Annotationen, die querschnittliche Aspekte konfigurieren (z.B. `@EnableTransactionManagement`), werden an die Konfigurationsklasse der inhaltlich passenden Schicht geschrieben.
+
+[[listing-konfigurationsklasse]]
+.Konfigurationsklasse des Anwendungskerns
+[source,java]
+----
+package de.beispiel.if2anwendung.core;
+
+@Configuration
+@EnableTransactionManagement
+@ComponentScan
+public class CoreConfig {
+
+    @Bean
+    public BeispielBean beispielBean() {
+        // ...
+    }
+}
+----
+
+Die Spring-Beans der Komponenten werden mit der Annotation `@Component` bzw. deren Spezialisierungen (`@Repository`, `@Service`, `@Controller` etc.) versehen.
+Externe Spring-Beans, die nicht annotiert werden können, werden in der Konfigurationsklasse der Schicht per `@Bean`-Methoden konfiguriert.
+
+[[webapprootkey-konfigurieren]]
+=== webAppRootKey konfigurieren
+
+Der Kontextparameter `webAppRootKey` muss in der `web.xml` auf den eindeutigen Namen der Webanwendung gesetzt werden.
```

Comment:
und jetzt sind wir plötzlich in die web.xml hinein gestolpert ... 
und für diese xml-Datei wird nur eine klitzekleine Vorgabe gemacht ...
Ich würde vermuten, dass zur web.xml Datei vielleicht noch mehr zu sagen wäre 
Gleichzeitig fällt auf, dass zur application.yml keinerlei Vorgaben gemacht werden

Insgesamt heißt das, dass am Anfang des Dokuments viel klarer der Scope 
des Dokuments  definiert und im weiteren Verlauf auch eingehalten werden sollte.

Andernfalls wird dieses Dokument über die Zeit noch um tausend Aspekte ergänzt werden
und man verliert die Struktur

Reply:
Für die Überarbeitung dieses Dokument gibt es bereits ein Ticket. Ich habe einen Kommentar ergänzt.